api/deployment/list.go

```diff
@@ -125,7 +122,7 @@ func ListDeployments(c *gin.Context) {
 	}
 
 	// send API call to capture the list of deployments for the repo
-	d, err := scm.FromContext(c).GetDeploymentList(ctx, u, r, page, perPage)
+	d, err := database.FromContext(c).ListDeployments(c)
```

Comment:
This should be `ListDeploymentsForRepo`. We don't want the entire database table returned on this request